### PR TITLE
add stream

### PR DESCRIPTION
**Pull request by super626**
add stream and element node

### CODE REVIEW

File: cocos/3d/ElementNode.cpp [hunk 1]
```diff
@@ -0,0 +1,1102 @@
...
+
+    do
+    {
+		stream->read(&c);
+    } while (isspace(c) && c!=EOF);
+
+    // If we are not at the end of the file, then since we found a
+    // non-whitespace character, we put the cursor back in front of it.
+    if (c != EOF)
+	{
```
Comment (minggo) [inline cocos/3d/ElementNode.cpp:27]: Wrong indent.

File: cocos/3d/ElementNode.cpp [hunk 2]
```diff
@@ -0,0 +1,1102 @@
...
+		if (stream->seek(-1, SEEK_CUR) == false)
+        {
+            CCAssert(0, "Failed to seek backwards one character after skipping whitespace.");
+        }
+	}       
+}
+
+char* trimWhiteSpace(char *str)
+{
+    if (str == NULL)
```
Comment (minggo) [inline cocos/3d/ElementNode.cpp:37]: NULL -> nullptr.

**Comment by super626** [comment]
sorry, close it. I will repull it

File: cocos/3d/ElementNode.cpp [hunk 3]
```diff
@@ -0,0 +1,1102 @@
...
+    return _nodeType.c_str();
+}
+
+const char* ElementNode::getNodeName() const
+{
+    return _nodeName.c_str();
+}
+
+
+const bool isStringNumeric(const char* str)
```
Comment (minggo) [inline cocos/3d/ElementNode.cpp:670]: bool value don't have to add const. Did you want to add it at the end of function?

**Comment by minggo** [comment]
And there are conflicts of this pull request.

**Pull request closed without merging**